### ISSUE TITLE
Unimplemented, Unmapped, Optional Protocol Properties crash MTLModel -validate:

### ISSUE THREAD

**Issue by kevin-traansmission** [Mantle/Mantle#523]
(Mantle v2.0, iOS 8.3, Xcode 6.3.1)

When I define a protocol with an optional property, and I do not implement that property in my MTLModel subclass, and do not define the property by name in +JSONKeyPathsByPropertyKey, 

Then I'm seeing a crash in MTLModel -validate:

**\* Terminating app due to uncaught exception 'NSUnknownKeyException', reason: '[<Foo 0x7f9582d81280> valueForUndefinedKey:]: this class is not key value coding-compliant for the key bar.'

I'll try to make a sample project that shows this.

**Comment by kevin-traansmission** [comment]
Sample project [here](https://github.com/kevin-traansmission/MantleOptionalProtocolProperty)

**Comment by jspahrsummers** [comment]
Thanks for the bug report!

**Comment by kevin-traansmission** [comment]
Logically, what we need is have MTLModel +storageBehaviorForPropertyWithKey: return MTLPropertyStorageNone for an unimplemented, optional protocol property in the model class.

I did some exploring and found a bug in the Obj-C runtime :-( [rdar:21065389](https://openradar.appspot.com/21065389). I can send you the sample project I uploaded to Apple if you want to duplicate my radar.

Since we can't determine if the property is optional in the protocol, I think the next best thing would be to check the following:
* mtl_propertyAttributes dynamic should be nil
* mtl_propertyAttributes ivar should be NULL
* class_getInstanceMethod( class, mtl_propertyAttributes getter ) should be NULL
* class_getInstanceMethod( class, mtl_propertyAttributes setter ) should be NULL

If this sounds correct, I'll send a pull request.

Thanks,
-kevin
